docs/mp/openapi/01_openapi.adoc

```diff
@@ -182,15 +160,241 @@ include::{common-openapi-doc}[common-config]
 There is not yet an adopted IANA YAML media type, but a proposed one specifically
 for OpenAPI documents that has some support is `application/vnd.oai.openapi`.
 That is what Helidon returns, by default.
 
-In addition a client can specify the HTTP header `Accept:` as either `application/vnd.oai.openapi+json` or `application/json`
+In addition, a client can specify the HTTP header `Accept:` as either `application/vnd.oai.openapi+json` or `application/json`
 to request JSON. Alternatively, the client can pass the query parameter `format` as either `JSON` or `YAML` to receive `application/json` or `application/vnd.oai.openapi` (YAML) output, respectively.
+
+== Configuration
+
+Helidon OpenAPI configuration is supported by the following class:
+
+include::../../shared/config/io.helidon.openapi.SEOpenAPISupport.adoc[]
+
+=== Helidon MP OpenAPI configuration.
+
+include::../../shared/config/io.helidon.openapi.internal.OpenAPIConfigImpl.Builder.adoc[]
+
+=== Smallrye Configuration
+
+Helidon MP OpenAPI implementation is based on link:https://smallrye.io/[Smallrye]. The configuration is backed by the following class:
+include::../../shared/config/io.smallrye.openapi.api.OpenApiConfig.adoc[]
+
+=== Configuration Example
+
+The classes with filters should be specified in `META-INF/microprofile-config.properties`:
+
+.Example Configuration
+[source,properties]
+----
+mp.openapi.filter=io.helidon.microprofile.examples.openapi.basic.internal.SimpleAPIFilter
+mp.openapi.model.reader=io.helidon.microprofile.examples.openapi.basic.internal.SimpleAPIModelReader
+----
+
+
+== Examples
+
+=== Helidon MP Basic OpenAPI Example
+
+This example shows a simple greeting application, similar to the one from the
+Helidon MP QuickStart, enhanced with OpenAPI support.
+
+[source,java]
+----
+@Path("/greeting")
+@PUT
+@Operation(summary = "Set the greeting prefix",
+           description = "Permits the client to set the prefix part of the greeting (\"Hello\")") <1>
+@RequestBody( <2>
+    name = "greeting",
+    description = "Conveys the new greeting prefix to use in building greetings",
+    content = @Content(
+                mediaType = "application/json",
+                schema = @Schema(implementation = GreetingMessage.class),
+                examples = @ExampleObject(
+                    name = "greeting",
+                    summary = "Example greeting message to update",
+                    value = "New greeting message")))
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public Response updateGreeting(JsonObject jsonObject) {
+    ...
+}
+----
+1. With `@Operation` annotation we document the current method.
+2. With `@RequestBody` annotation we document the content produced. Internal annotations `@Content`, `@Schema` and `@ExampleObjects` are used to give more details about the returned data.
+
+If we want to hide a specific path an `OASFilter` is used:
```

Comment:
Filters are more general than this. Maybe steal a small part of the MP OpenAPI doc text from here https://download.eclipse.org/microprofile/microprofile-open-api-2.0/microprofile-openapi-spec-2.0.html#_filter

Then add:
> The following example filter prevents information about a given path from appearing in the OpenAPI document.

Reply:
Looks good, although maybe remove the `NOTE:` introducer. This can just be part of the regular explanatory text about the filter API.

Reply:
Removed NOTE: